data/org.freedesktop.portal.InputCapture.xml

```diff
@@ -0,0 +1,547 @@
+        implies the application is not using the current zone set and
+        pointer barriers will fail.
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>zones a(uuii)</term>
+            <listitem><para>
+              An array of retions, each specifying that zone's width,
+              height, x/y offset.
+            </para></listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>serial u</term>
+            <listitem><para>
+             A serial number to be used in the
+             org.freedesktop.portal.InputCapture.SetPointerBarriers() method to refer to
+             this set of zones. This serial increases by an unspecified
+             amount whenever the zones change and pointer barriers can only be set up
+             if the serial matches the most recent retured serial.
+
+             Note that this serial has no relation to the serial in
+             org.freedesktop.portal.InputCapture.Enable()
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+    -->
+    <method name="GetZones">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="handle" direction="out"/>
+    </method>
+
+    <!--
+        SetPointerBarriers:
+        @session_handle: Object path for the #org.freedesktop.portal.Session object
+        @options: Vardict with optional further information
+        @barriers: An array of vardicts, each specifying one barrier
+        @serial: A unique serial number referring to the zone  set
+        @handle: Object path for the #org.freedesktop.portal.Request object representing this call
+
+        Set up zero or more pointer barriers. Pointer barriers are horizontal
+        or vertical lines that should trigger the start of input capture when the cursor moves
+        across the pointer barrier. In the future,
+        when the compositor has deemed the pointer barrier to be crossed, it may
+        complete the org.freedesktop.portal.InputCapture.Enable() request and input events
+        (from compositor-determined input devices) are sent to the application
+        via the transport layer.
+
+        Note that unlike the X11 XFixes PointerBarriers, these pointer barriers do not
+        block movement of the cursor.
+
+        Pointer barriers are situated on the top or left of their respective pixels and a
+        pointer barrier must be within or at the extent of the available
+        zone. For example, a zone of 1920x1080 at offset 0,0 allows
+        a pointer barrier at x1=1920, y1=0, x2=1920, y2=1080.
+        A pointer barrier at the extent of a zone is considered triggered
+        when the pointer would logically move off that zone, even if the
+        actual cusor movement is clipped to the zone.
```

Comment:
Should we limit these barriers on the outer bounds of all the zones? Or should it be possible to create a barrier in between two adjecent zones?

Also what about barriers right in the middle? Sounds like something that shouldn't be possible, since the application cannot possibly know what is there.

Also, should it be allowed that one barrier can "touch" multiple zones? I.e. if you have

```
+---+---+
| 1 | 2 |
+---+---+
A   B   C
```
Can a barrier be created between A and C, or should it be two barriers, one between A and B, and another between B and C?

Reply:
Agree with the outer-bound zones-only, that's (at least for version 1) a sensible approach.

And I think requiring A-B and B-C separately makes sense. There's a use-case for having them separate anyway (synergy may have two different clients "above" 1 and 2) and it makes the checks a bit simpler when 1 and 2 are of different sizes/offsets.

Reply:
done in the latest updates, direction was removed as well